app/server/lib/DocApi.ts

```diff
@@ -538,6 +542,36 @@ export class DocWorkerApi {
       });
     }));
 
+    // Returns the status of any current / pending attachment transfers
+    this._app.get('/api/docs/:docId/attachments/store', isOwner,
+      withDoc(async (activeDoc, req, res) => {
+        const storeId = await activeDoc.getAttachmentStore();
+        res.json({
+          type: storeId? 'external' : 'internal',
+        });
+      })
+    );
+
+    this._app.post('/api/docs/:docId/attachments/store', isOwner, validate(SetAttachmentStorePost),
+      withDoc(async (activeDoc, req, res) => {
+        const body = req.body as Types.SetAttachmentStorePost;
+        if (body.type === 'internal') {
+          await activeDoc.setAttachmentStoreFromLabel(docSessionFromRequest(req), undefined);
+        }
+
+        if (body.type === 'external') {
+          const storeLabel = getConfiguredStandardAttachmentStore();
+          if (storeLabel === undefined) {
+            throw new ApiError("server is not configured with an external store", 400);
+          }
+          // This store might not exist - that's acceptable, and should be handled elsewhere.
+          await activeDoc.setAttachmentStoreFromLabel(docSessionFromRequest(req), storeLabel);
+        }
+
+        res.json(null);
```

Comment:
Can you be a bit more chatty in the reply? I know we do nulls in a lot of other places but it is a mistake. AIs trying to use the API feel very unsure what happened so I can only imagine people do too.

Reply:
I made it return the store ID of the document instead, so the user can know what the newly active store is.